src/margo-rpc-lineage.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * (C) 2024 The University of Chicago
+ *
+ * See COPYRIGHT in top-level directory.
+ */
+#ifndef __MARGO_ABT_KEY_H
+#define __MARGO_ABT_KEY_H
+#include <assert.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <string.h>
+#include <abt.h>
+#include <mercury.h>
+
+#define __MARGO_LINEAGE_COMMON                                    \
+    static const char* magic  = "matthieu";                       \
+    ABT_thread_attr attr      = ABT_THREAD_ATTR_NULL;             \
+    ABT_thread      ult       = ABT_THREAD_NULL;                  \
+    void*           stackaddr = NULL;                             \
+    size_t          stacksize = 0;                                \
+    int             ret;                                          \
+    ret = ABT_thread_self(&ult);                                  \
+    if(ret != ABT_SUCCESS) return ret;                            \
+    ret = ABT_thread_get_attr(ult, &attr);                        \
+    if(ret != ABT_SUCCESS) return ret;                            \
+    ret = ABT_thread_attr_get_stack(attr, &stackaddr, &stacksize);\
+    if(ret != ABT_SUCCESS) return ret;                            \
+    ABT_thread_attr_free(&attr);                                  \
+    if(!stackaddr || !stacksize) return ABT_ERR_KEY;              \
+    char* stackend = (char*)stackaddr + stacksize
+
+static inline int margo_lineage_set(hg_id_t current_rpc_id) {
+    __MARGO_LINEAGE_COMMON;
+    memcpy(stackend - 8 - sizeof(hg_id_t), magic, 8);
```

Comment:
Minor stylistic comment.  It looks odd for one sizeof() to use the type and the other sizeof() to use the variable (visually it makes you think that they are different values).  Maybe just use the same style all the way through?  I usually prefer the latter but the former might make more sense here since erase() doesn't have a variable name to use.